app/models/export.rb

```diff
@@ -18,62 +15,32 @@ class UnknownExportTypeError < StandardError; end
   scope :completed, -> { where.not(completed_at: nil) }
   scope :application_ids, -> { pluck(:snap_application_id) }
   scope :latest, -> { first }
-
-  def self.enqueue_faxes
-    SnapApplication.faxable.untouched_since(DELAY_THRESHOLD.minutes.ago).
-      find_each do |snap_application|
-      create_and_enqueue(destination: :fax, snap_application: snap_application)
-    end
-  end
-
-  def self.create_and_enqueue(params)
-    unless params[:snap_application].
-        exports&.
-        for_destination(params[:destination])&.
-        any?
-      create(params).enqueue
-    end
-  end
-
-  def self.create_and_enqueue!(params)
-    create!(params).enqueue
-  end
-
-  # Queues up the export job for the given destination
-  def enqueue
-    transaction do
-      save!
-      case destination
-      when :fax
-        FaxApplicationJob.perform_later(export: self)
-      when :email
-        EmailApplicationJob.perform_later(export: self)
-      when :sms
-        ApplicationSubmittedSmsJob.perform_later(export: self)
-      else
-        raise UnknownExportTypeError, destination
-      end
-      transition_to(new_status: :queued)
-    end
-  end
+  scope :without, ->(export) { where.not(id: export.id) }
+  scope :successful_or_in_flight, -> {
+    where(status: %i(new queued in_process
+                     succeeded))
+  }
 
   def execute
     raise ArgumentError, "#export requires a block" unless block_given?
 
-    if snap_application.exports.succeeded.
-        for_destination(destination).present? && !force
+    if snap_application.exports.for_destination(destination).
+        successful_or_in_flight.without(self).present? && !force
 
-      transition_to new_status: :failed
-      update(metadata: "Export failed because a previous export succeeded",
+      transition_to new_status: :unnecessary
+      update(metadata: "Another export for this destination and app occurred",
```

Comment:
this metadata message is a bit confusing to me. Maybe this is related to me not understanding when a status of `unnecessary` would occur

Reply:
Good point. Going to rephrase it to include more useful information even though it makes it a bit longer.